### DISCUSSION TITLE
Importing RootStore instead of Provider

### DISCUSSION THREAD

**Discussion by gravitymedianet** [mobxjs/mobx#2677] [Q&A] [answered]
Earlier in mobx 4-5 it was recommended to wrap your app component in a provider.

I want to know if it is possible to simply import the RootStore into any component that may need it instead of using the provider/inject method.

**Comment by danielkcz** [comment]
Please read carefully through: https://mobx.js.org/react-integration.html#local-and-external-state

**Reply by gravitymedianet** [reply]
For others searching for this, I believe FredyC is referring to the tab called "using global variables" quoted below:

"Since it doesn't matter how we got the reference to an observable, we can consume observables from outer scopes directly (including from imports, etc.):

"const myTimer = new Timer() // See the Timer definition above.

// No props, `myTimer` is directly consumed from the closure.
const TimerView = observer(() => <span>Seconds passed: {myTimer.secondsPassed}</span>)

"ReactDOM.render(<TimerView />, document.body)
Using observables directly works very well, but since this typically introduces module state, this pattern might complicate unit testing. Instead, we recommend using React Context instead."

**Comment by gravitymedianet** [comment]
Thanks, I do not see that my issue is covered.

What I would like to do is simply export default new RootStore from my
RootStore file and then import it into my observer components.

Is there a reason to not do this? It seems to work ok.

**Reply by danielkcz** [reply]
I said READ CAREFULLY. You have explicit mention of global variables there. The documentation is important to really read, not skim through the lines. You will thank me later.

**Comment by kubk** [comment, marked as answer]
From my experience, stores-singletons could be used in case you don't have SSR and don't write unit tests. With SSR your store will be shared among all the users (because the server starts only once and your store stays the same in module between different users). For unit tests, it may be difficult to always restore the store state between test cases. Again, because the instance of a global store is shared among test cases. In other cases, it works OK.

**Reply by gravitymedianet** [reply]
Thank you so much for taking the time to reply. So much more helpful that being told to "read documentation." I never even thought of these scenarios being an issue.

**Reply by danielkcz** [reply]
You know, that's what's documentation is for ... to read it. People made a huge effort to write it all down and you feel like it's easier to just shoot random questions without any code examples. I get that, but a bit more respect toward the work done here would be great.

**Reply by gravitymedianet** [reply]
You know, not everyone is an experienced developer. I am very grateful for the work but I find that the team can be quite rude some times. I have read that page and did not see the global section which is hidden away on a third tab. Perhaps making it more obvious would help. Remember, you are making this code to be used by others--as someone who maintains a website elsewhere, I always make it a point to respond kindly to my followers. They are, after all, taking a moment to show interest in my work and to communicate with me.

**Reply by danielkcz** [reply]
@gravitymedianet I do kindly respond to people who are willing to show an effort. If you just shoot such a generic question, it seems obvious you did not bother to do research on your own. We are not here to provide premium support. You are probably paid for the work you do, we are not.

And if you find documentation confusing or missing something, it's always better to open an issue about it instead of complaining about it. Or even better, contribute an improvement!

**Reply by gravitymedianet** [reply]
Asking a question IS contributing an improvement. People will search this and this will appear. And now there is a thread to provide deeper insight. BE KIND. You never know who you are talking to online.
